### PR TITLE
add ok robot and the third querying strategy

### PR DESCRIPTION
**Pull request by hello-peiqi**
## Description

Provide a brief summary of the changes introduced in this pull request.

## Checklist

- \[ \] I have performed a self-review of my code
- \[ \] If it is a core feature, I have added thorough tests
- \[ \] I have added documentation for the changes
- \[ \] I have updated the README file if necessary
- \[ \] I have run on hardware if necessary

## Screenshots (if applicable)

Add any relevant screenshots or screen recordings to help reviewers understand the changes.

## Additional context

Add any other context or information about the pull request here.

### CODE REVIEW

**Review by cpaxton — APPROVED** [review]
lgtm

File: src/stretch/dynav/mapping_utils/voxel.py [hunk 1]
```diff
@@ -400,8 +400,8 @@ def add(
                 return
 
             # Remove points that are too close to the robot
-            mesh = self.urdf_visualizer.get_combined_robot_mesh()
-            selected_indices = points_in_mesh(world_xyz, mesh)
+            # mesh = self.urdf_visualizer.get_combined_robot_mesh()
```
Comment (hello-cpaxton) [inline src/stretch/dynav/mapping_utils/voxel.py:403]: Should this be removed? What's it for?

File: src/stretch/dynav/voxel_map_server.py [hunk 2]
```diff
@@ -629,10 +627,10 @@ def process_rgbd_images(self, rgb, depth, intrinsics, pose):
                 camera_K=torch.Tensor(intrinsics),
             )
 
-        if self.image_shape is not None:
-            rgb = F.interpolate(
-                rgb.unsqueeze(0), size=self.image_shape, mode="bilinear", align_corners=False
-            ).squeeze()
+        # if self.image_shape is not None:
```
Comment (hello-cpaxton) [inline src/stretch/dynav/voxel_map_server.py:630]: Why is this removed?

File: src/stretch/llms/openai_client.py [hunk 3]
```diff
@@ -39,7 +39,7 @@ def __init__(
         ), f"model must be one of {self.model_choices}, got {self.model}"
         self._openai = OpenAI()
 
-    def __call__(self, command: str, verbose: bool = False):
+    def __call__(self, command: Union[str, list], verbose: bool = False):
         # prompt = copy.copy(self.prompt)
```
Comment (hello-cpaxton) [inline src/stretch/llms/openai_client.py:43]: dead code?

File: src/stretch/mapping/voxel/voxel_dynamem.py [hunk 4]
```diff
@@ -408,6 +418,179 @@ def add_to_semantic_memory(self, valid_xyz, feature, valid_rgb, weights=None, th
...
 
     def localize_A(self, text, debug=True, return_debug=False):
+        if self.mllm:
+            return self.localize_A_with_mllm(text, debug=debug, return_debug=return_debug)
+        else:
+            return self.localize_with_feature_similarity(
+                text, debug=debug, return_debug=return_debug
+            )
+
+    def find_all_images(self, A):
```
Comment (hello-cpaxton) [inline src/stretch/mapping/voxel/voxel_dynamem.py:428]: Add type hints and docstring

File: src/stretch/mapping/voxel/voxel_dynamem.py [hunk 5]
```diff
@@ -408,6 +418,179 @@ def add_to_semantic_memory(self, valid_xyz, feature, valid_rgb, weights=None, th
...
+                return numbers[-1]
+            else:
+                return None
+
+        except Exception as e:
+            # Handle any exceptions and optionally log the error message
+            print(f"Error: {e}")
+            return None
+
+    def localize_A_with_mllm(self, A, debug=True, return_debug=False):
```
Comment (hello-cpaxton) [inline src/stretch/mapping/voxel/voxel_dynamem.py:549]: shoudl A be object, location, query...

**Review by hello-cpaxton — COMMENTED** [review]
mostly good but some minor comments/questions

File: src/stretch/dynav/voxel_map_server.py (hunk 2 above)
Comment (hello-peiqi) [reply src/stretch/dynav/voxel_map_server.py:630]: I moved it to above.

Comment (hello-peiqi) [reply src/stretch/dynav/voxel_map_server.py:630]: I moved it to above

File: src/stretch/llms/openai_client.py (hunk 3 above)
Comment (hello-peiqi) [reply src/stretch/llms/openai_client.py:43]: No, I am using it right now.

File: src/stretch/mapping/voxel/voxel_dynamem.py (hunk 5 above)
Comment (hello-peiqi) [reply src/stretch/mapping/voxel/voxel_dynamem.py]: Resolved

**Review by hello-cpaxton — APPROVED** [review]
LGTM
